child_process: document the error when cwd does not exist (#34505)
nodejs/node: doc/api/child_process.md

```

Use `cwd` to specify the working directory from which the process is spawned.
If not given, the default is to inherit the current working directory. If given,

> but the path does not exist, the child process will emit an `ENOENT` then
> exit immediately, **this is the same as the error emitted when the command
> does not exist**.

Reviewer: ```suggestion
but the path does not exist, the child process emits an `ENOENT` error
and exits immediately. `ENOENT` is also emitted when the command
does not exist.
```

I'm not sure "emit `ENOENT`" is right, but more accurate characterizations get pretty wordy, so maybe it's OK, at least for now.
